### PR TITLE
feat(multistream-select): remove Version

### PR DESCRIPTION
**Pull request by mxinden**
## Description

Removes multistream-select version selection. Instead the mechanism of `Version::V1Lazy` is always used.

See also https://github.com/libp2p/rust-libp2p/issues/3563 and https://github.com/libp2p/rust-libp2p/pull/3749.

## Notes & open questions

I suggest we deprecate `Version` before removing it. Thoughts?

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: core/tests/transport_upgrade.rs [hunk 1]
```diff
@@ -82,7 +82,7 @@ fn upgrade_pipeline() {
     let listener_keys = identity::Keypair::generate_ed25519();
     let listener_id = listener_keys.public().to_peer_id();
     let mut listener_transport = MemoryTransport::default()
-        .upgrade(upgrade::Version::V1)
+        .upgrade()
```
Comment (thomaseizinger) [inline core/tests/transport_upgrade.rs:85]: Technically we could remove `upgrade` now, right?

What do you think of deprecating `upgrade` first and instead offer an `authenticate` function for each transport that simply implies `upgrade(V1Lazy)`?

**Review by thomaseizinger — COMMENTED** [review]
Concept ACK

If you want to remove it in `v0.53`, then I'd suggest we ship a deprecation of `upgrade` as a back-port first. Perhaps we should be a bit cautious here though. So far, I think the `v0.53` change does not contain anything controversial whereas I remember that `V1Lazy` caused some issues in polkadot, right? Perhaps we should push it to the next breaking release and make sure polkadot upgrades to `v0.53` asap after this.

File: core/tests/transport_upgrade.rs (hunk 1 above)
Comment (jxs) [reply core/tests/transport_upgrade.rs:85]: Yeah, while looking into the PR this also made sense to me, would simplify the API, but would also probably make it a breaking change no?

Comment (thomaseizinger) [reply core/tests/transport_upgrade.rs:85]: At the moment it is a breaking change. Deprecating it first and offering a new API that starts with `authenticate` straight from `Transport` is not.

**Comment by mxinden** [comment]
I am in favor of deprecating `Transport::upgrade` and `multistream_select::Version` first. I don't think this needs to land in `v0.53`.

**Comment by mxinden** [comment]
Closing here. For anyone interested in this path, feel free to cherry-pick the commit.

**Pull request closed without merging**